src/ImageSharp/Configuration.cs

```diff
@@ -70,6 +94,32 @@ public void AddImageFormat(IImageFormat format)
         }
 
         /// <summary>
+        /// Tries the add image format.
+        /// </summary>
+        /// <param name="typeName">Name of the type.</param>
+        /// <returns>True if type discoverd and is a valid <see cref="IImageFormat"/></returns>
+        internal bool TryAddImageFormat(string typeName)
+        {
+            Type type = Type.GetType(typeName, false);
+            if (type != null)
+            {
+                IImageFormat format = Activator.CreateInstance(type) as IImageFormat;
+                if (format != null
+                    && format.Encoder != null
+                    && format.Decoder != null
+                    && !string.IsNullOrEmpty(format.MimeType)
+                    && format.SupportedExtensions?.Any() == true)
+                {
+                    // we can use the locked version as we have already validated in the if.
```

Comment:
@dlemstra these are non-throwing checks, while throws an exception when something is missing. I see no straightforward way to refactor this without catching first-chance exceptions on *normal* execution path.